aiida_vasp/utils/compare_bands.py

```diff
@@ -284,3 +284,75 @@ def plot_errors_vs_iwsize(comparison_info):
     lines = btool.plt.plot(iws_i, plot_data, figure=fig)
     btool.plt.legend(lines, ows_i)
     return fig, zip(ows, iws, data)
+
+
+def get_band_properties_from_data(bandsdata):
+    """
+    Acquire bands information from BandsData
+
+    Parameters
+    ----------
+    bandsdata : BandsData
+        A BandsData node to be processed
+
+    Returns
+    -------
+    bands_info : dict
+        Same output as get_band_properties
+    """
+
+    eigenvalues, occupations = bandsdata.get_bands(also_occupations=True)
+    return get_band_properties(eigenvalues, occupations)
+
+
+def get_band_properties(eigenvalues, occupations):
+    """
+    Get some auxiliary information about the band structure
+
+    Parameters
+    ----------
+    eigenvalues : np.ndarray
+        The eigenvalue array has shape (NS, NB, NK) or (NB, NK), where NS is the number of spins,
+        NB is the number of bands and NK is the number of kpoints.
+
+    occupations: np.ndarray
+        Same format as eigenvalues but instead holds occupation numbers.
+
+    Returns
+    -------
+    bands_info : dict
+        A dictionary holds information about VBM and CBM locations and bandgaps.
+
+    """
+    import numpy as np
+
+    info = {}
+    vbm = -float('inf')
+    cbm = float('inf')
+    vbm_kpt = None
+    cbm_kpt = None
+    # Add the spin dimension if it is missing
+    if len(eigenvalues.shape) == 2:
+        eigenvalues = eigenvalues.reshape((1,) + eigenvalues.shape)
+    if len(occupations.shape) == 2:
+        occupations = occupations.reshape((1,) + occupations.shape)
+
+    # NOTE: this may not correctly handle multiple degenerate VBM/CBM levels
+    for spin, occ in enumerate(occupations):
+        eign = eigenvalues[spin]
+        occupied = occ > 1e-8
```

Comment:
Sometimes occupations are negative, so we possibly need to have an abs here just to be sure.

Reply:
I think this is related to the tetrahedral method (ISMEAR=-5)?
If a band has a negative occupation number should it be counted as occupied or unoccupied?

Reply:
Yes, that is correct. We can have both, plus/minus, so we can have bigger then two and smaller than zero. I do not think we can determine if it should be part of the occupied of unoccupied band as it is just purely a numerical correction. Maybe one could do something fancy, but we do not have that framework in place.